internal/handlers/initrd.go

```diff
@@ -89,6 +95,30 @@ func initrdOverlayReader(imageStore imagestore.ImageStore, client *AssistedServi
 			return nil, "", http.StatusInternalServerError, fmt.Errorf("failed to create append reader for initrd: %v", err)
 		}
 
+		versionOK, err := common.VersionGreaterOrEqual(version, isoeditor.MinimalVersionForNmstatectl)
+		if err != nil {
+			return nil, "", http.StatusInternalServerError, err
+		}
+
+		if versionOK {
+			extractDir, err := os.MkdirTemp(workDir, "isoutil")
+			if err != nil {
+				return nil, "", http.StatusInternalServerError, err
+			}
+			defer os.RemoveAll(extractDir)
+			if err = isoeditor.Extract(isoPath, extractDir); err != nil {
+				return nil, "", http.StatusInternalServerError, err
+			}
+			rootfsPath := filepath.Join(extractDir, "images/pxeboot/rootfs.img")
+			nmstateCpioArchive, err := nmstateHandler.BuildNmstateCpioArchive(rootfsPath)
```

Comment:
So IIUC, it means that we're extracting the binary and building the archive for every '/pxe-initrd' request.
I.e. we probably need to consider caching of the binary. Maybe we could simply save the cpis in the ImageStore when handling the templates? It would also simply the logic here. Instead of rebuilding, just read it from the ImageStore, e.g. something like: https://github.com/openshift/assisted-image-service/blob/1362d5b6f4e729ab797704bd5c48535dc2192a4e/internal/handlers/initrd.go#L75

Reply:
Yeah I wonder what would happen if we got a ton of those requests at the same time.
Probably nothing good.